src/python/pants/backend/visibility/rule_types_test.py

```diff
@@ -193,33 +201,53 @@ def test_visibility_rule_set_match(expected: bool, target: str, rule_spec: tuple
 @pytest.fixture
 def dependencies_rules() -> BuildFileVisibilityRules:
     return BuildFileVisibilityRules(
-        "test/BUILD",
+        "a/BUILD",
         # Rules for outgoing dependency.
-        (parse_ruleset(("*", ("tgt/ok/*", "?tgt/dubious/*", "!tgt/blocked/*"))),),
+        (parse_ruleset(("*", ("tgt/ok/*", "?tgt/dubious/*", "!tgt/blocked/*")), "a/BUILD"),),
     )
 
 
 @pytest.fixture
 def dependents_rules() -> BuildFileVisibilityRules:
     return BuildFileVisibilityRules(
-        "test/BUILD",
+        "b/BUILD",
         # Rules for outgoing dependency.
-        (parse_ruleset(("*", ("src/ok/*", "?src/dubious/*", "!src/blocked/*"))),),
+        (parse_ruleset(("*", ("src/ok/*", "?src/dubious/*", "!src/blocked/*")), "b/BUILD"),),
     )
 
 
 @pytest.mark.parametrize(
-    "source_path, target_path, expected_action",
+    "source_path, target_path, expected_action, expected_rule",
     [
-        ("src/ok/a", "tgt/ok/b", "allow"),
-        ("src/ok/a", "tgt/dubious/b", "warn"),
-        ("src/ok/a", "tgt/blocked/b", "deny"),
-        ("src/dubious/a", "tgt/ok/b", "warn"),
-        ("src/dubious/a", "tgt/dubious/b", "warn"),
-        ("src/dubious/a", "tgt/blocked/b", "deny"),
-        ("src/blocked/a", "tgt/ok/b", "deny"),
-        ("src/blocked/a", "tgt/dubious/b", "deny"),
-        ("src/blocked/a", "tgt/blocked/b", "deny"),
+        ("src/ok/a", "tgt/ok/b", "allow", "a/BUILD[tgt/ok/*] -> b/BUILD[src/ok/*]"),
```

Comment:
uhm.... the direction is swapped here. 😬 will fix.

Reply:
Ah, no. They're not. But the test path names for the BUILD files could be improved, to make things clearer.